### PR TITLE
added teamcity example

### PR DESCRIPTION
**Pull request by hecomp**
## Description
What does this PR do?

## PR checklist for examples

Check these items before merging the PR.

- [x] Each example contains a README file.
- [x] README documents all steps and variables required to run the example. The simplest way to confirm this is to follow the instructions on a clean machine.
- [x] README file contains the links to the official documentation with relevant configuration and to the metrics published if that is available.
- [ ] Used images of apps and collectors have to be pinned, "latest" should never be used.
- [ ] Evidences with screenshots provided from the reviewer.
- [ ] Includes a file called `metrics.csv` with metrics produced by the example.
    The file should have these 5 headings: Name, Description, Unit, DataType, Attributes.
    Description is not provided for all metrics, so it may be blank. Attributes are also not always provided. When there are multiple attributes on a metric reord they should be space separated.
- [ ] Dashboard provided with example and put to the folder "dashboard" inside the example folder.
- [ ] At least one screenshot of the proposed dashboard is included in this PR. If you're proposing substantive changes to queries or new queries then please ensure your screenshot shows relevant data.
- [ ] All chart names in the dashboard are in [Title case](https://en.wikipedia.org/wiki/Title_case).
- [ ] All query names in the dashboard are lower case letters, beginning the first query of each chart as "a" and proceeding alphabetically ... "b", "c", etcetera.

### CODE REVIEW

File: collector/teamcity/examples/compose/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - ./data/teamcity_server/datadir:/data/teamcity_server/datadir
+      - ./logs:/opt/teamcity/logs
+
+  teamcity-agent:
+    container_name: teamcity-agent
+    image: jetbrains/teamcity-agent:latest
+    environment:
+      - SERVER_URL=http://teamcity-server:8111
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock
```
Comment (nslaughter) [inline collector/teamcity/examples/compose/docker-compose.yml:18]:
```suggestion
```
No need to mount docker socket for scraping a prometheus endpoint

**Review by nslaughter — CHANGES_REQUESTED** [review]
@hecomp 
It's not working state yet. Note that this is only producing metrics from the Prometheus receiver and that the scrapes are failing (see the logs). There are a couple of things I would try after a quick look.

Needs some additional config or possibly walk through of settings to expose the metrics at 8111/app/metrics

File: collector/teamcity/examples/compose/docker-compose.yml (hunk 1 above)
Comment (hecomp) [reply collector/teamcity/examples/compose/docker-compose.yml:18]: Hmmm ok. Make sense.

**Comment by hecomp** [comment]
> @hecomp It's not working state yet. Note that this is only producing metrics from the Prometheus receiver and that the scrapes are failing (see the logs). There are a couple of things I would try after a quick look.
> 
> Needs some additional config or possibly walk through of settings to expose the metrics at 8111/app/metrics

I guess I'm missing a small thing here.

**Comment by nslaughter** [comment]
Moved this to a draft state. We can revisit when comments are fully addressed.

**Pull request closed without merging**